domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/calendardates/CalendarDate.java

```diff
@@ -158,4 +159,29 @@ public CalendarDateBuilder clear() {
             return this;
         }
     }
+
+    /**
+     * Generates an hash code for this {@link CalendarDate} based on service_id and date fields
+     *
+     * @return an hash code for this {@link CalendarDate} based on service_id and date fields
+     */
+    @Override
+    public int hashCode() {
+        return Objects.hash(serviceId.hashCode(), date.hashCode());
```

Comment:
While this might technically work for our current purposes of enforcing uniqueness of serviceId and date combinations, after looking at this I think it's a dangerous change that could result in strange bugs later. It would be easy for a future developer to call `CalendarDate.equals()` and assume that two instances of `CalendarDate` that have the same serviceId and date but different exception type would return `false`. But with this implementation it would return `true`.

Let's stick with the existing `HashMap` impl in the GTFS repo instead of the new `Set` that relies on this for now, and we can revisit later in #360.